src/parsers/manifest/local/parse_local_manifest.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import idGenerator from "../../../utils/id_generator";
+import {
+  IParsedAdaptation,
+  IParsedManifest,
+  IParsedPeriod,
+  IParsedRepresentation,
+} from "../types";
+import createRepresentationIndex from "./representation_index";
+import {
+  ILocalAdaptation,
+  ILocalManifest,
+  ILocalPeriod,
+  ILocalRepresentation,
+} from "./types";
+
+const generateManifestID = idGenerator();
+
+/**
+ * @param {Object} localManifest
+ * @returns {Object}
+ */
+export default function parseLocalManifest(
+  localManifest : ILocalManifest
+) : IParsedManifest {
+  if (localManifest.type !== "local") {
+    throw new Error("Invalid local manifest given. It misses the `type` property.");
+  }
+  if (localManifest.version !== "0.1") {
+    throw new Error(`The current Local Manifest version (${localManifest.version})` +
+                    " is not compatible with the current version of the RxPlayer");
+  }
+  const periodIdGenerator = idGenerator();
+  const { isFinished } = localManifest;
+  const manifest = {
+    availabilityStartTime: 0,
+    duration: localManifest.duration,
+    expired: localManifest.expired,
+    id: "local-manifest_" + generateManifestID(),
+    transportType: "local",
+    isDynamic: !localManifest.isFinished,
+    isLive: false,
+    uris: [],
+    periods: localManifest.periods
+      .map(period => parsePeriod(period, periodIdGenerator, isFinished)),
+  };
+  return manifest;
+}
+
+/**
+ * @param {Object} period
+ * @returns {Object}
+ */
+function parsePeriod(
+  period : ILocalPeriod,
+  periodIdGenerator : () => string,
+  isFinished : boolean
+) : IParsedPeriod {
+  const adaptationIdGenerator = idGenerator();
+  return {
+    id: "period-" + periodIdGenerator(),
+    start: period.start,
+    end: period.duration - period.start,
+    duration: period.duration,
+    adaptations: period.adaptations
+      .reduce<Partial<Record<string, IParsedAdaptation[]>>>((acc, ada) => {
+        const type = ada.type;
+        if (acc[type] == null) {
```

Comment:
Do you still use `loose` equality? instead of `strict`?

Reply:
Nope, true that

Reply:
Thanks, done.